### PR TITLE
replace PUSH() Macro with pushback

### PR DESCRIPTION
**Pull request by jokoho48**

### CODE REVIEW

File: addons/common/fnc_getGroupIndex.sqf [hunk 1]
```diff
@@ -42,7 +42,7 @@ _number = [];
 // Format of player label is "<groupName>:<groupNumber> <playerName>"
 for "_i" from (_groupLabelLen + 1) to ((count _labelArray) - 1) do {
     if ((_labelArray select _i) == ASCII_SPACE) exitWith {};
-    PUSH(_number,_labelArray select _i);
+    _number pushBack _labelArray select _i;
```
Comment (commy2) [inline addons/common/fnc_getGroupIndex.sqf]: `_number pushBack (_labelArray select _i);

File: addons/common/XEH_postInit.sqf [hunk 2]
```diff
@@ -96,7 +96,7 @@ if !(isDedicated) then {
                 [GVAR(actionList), {
                     TRACE_3("Inside the code for the hashPair",(vehicle player),GVAR(actionIndexes), _value);
                     if (!isNil "_value" && typeName(_value) == "ARRAY") then {
-                        PUSH(GVAR(actionIndexes), (vehicle player) addAction _value)
+                        GVAR(actionIndexes) pushBack (vehicle player) addAction _value
```
Comment (Killswitch00) [inline addons/common/XEH_postInit.sqf]: Does this need safeguarding with
` GVAR(actionIndexes) pushBack ((vehicle player) addAction _value)` ?

Comment (esteldunedain) [inline addons/common/XEH_postInit.sqf]: When in doubt its better to add it anyway

Comment (commy2) [inline addons/common/XEH_postInit.sqf]: Yeah. This is missing the brackets too.
Evaluations of binary commands are done from left to right:
`GVAR(actionIndexes) pushBack (vehicle player) addAction _value`
->
`<index/number> addAction _value`
->
`Error Number exected object`

The brackets around `(vehicle player)` on the other hand are optional.

Correct and minimalistic:
`GVAR(actionIndexes) pushBack (vehicle player addAction _value)

I personally think that it's best to use as few brackets as possbile. Only exception would be
`(unaryCommand _var1) binaryCommand _var2`, so in the end exactly what killswitch proposed.

Comment (esteldunedain) [inline addons/common/XEH_postInit.sqf]: > I personally think that it's best to use as few brackets as possbile

I know you do :wink:.

Just to ring the opposite bell, I think it's usefull for the code to be as self-obvious as possible, so reviewers/maintainers don't have to wonder if a certain line is being interpreted correctly or not.

Comment (commy2) [inline addons/common/XEH_postInit.sqf]: >I think it's usefull for the code to be as self-obvious as possible
True. But everyone reviewing SQF should know the basic principles. And too many brackets hurt readability.
That discussion doesn't belong here though. The line is currently wrong and errors out.

**Comment by jokoho48** [comment]
ok change/fixed

**Comment by commy2** [comment]
:+1:
looks good

**Pull request merged**